### ISSUE TITLE
Send proper Error message on API OutOfMemoryError

### ISSUE THREAD

**Issue by SlowMo24** [GIScience/ohsome-api#185]
### Use Case Description
Too large bpolys can overcharge the ohsome api depending on its settings (e.g. with a 3MB geojson of 900 polygons with an avg of 109 vertices). Sending this request will (after some time) return the following "error":

```
    "status": "200,\n  \"error\" : \"OK\""
    "message": "java.lang.OutOfMemoryError\",\n  \"path\" : \"/elements/geom"
```

This message is confusing (error and OK at the same time) and does not help the user how to overcome this issue or what went wrong.

### Request Description
The returned message should be improved, even though it may not be cause by the ohsome-api itself but by the used framework.

ohsome-py logs: [errorOnLargeBuffer.zip](https://github.com/GIScience/ohsome-api/files/6440767/errorOnLargeBuffer.zip)

**Comment by tyrasd** [comment]
> error and OK at the same time

at first glance, this seems like a bug from your description. what kind of request endpoint was this? do you have a way to share the large input geometry and rest of the query parameters, so we can replicate this issue?

**Comment by SlowMo24** [comment]
> could you please share the large input geometry and rest of the query parameters, so we can replicate this issue?

I updated the issue with a file link.

**Comment by tyrasd** [comment]
rest of query also, please :pray: (see https://github.com/GIScience/ohsome-api/blob/master/.github/ISSUE_TEMPLATE/bug_report.md#general-information)

**Comment by tyrasd** [comment]
one last question: is the snippet above the full result you get, or is there more which you omitted?

**Comment by tyrasd** [comment]
OK, this is a data extraction request, which could be a first hint why the status code is 200 "OK": these requests use chunked/streaming data output, where there is no technical way to modify the status code from 200 to something else once data transmission has begun.

Anyway, I agree that the error message should not contain `"error" = "OK"` since it is highly confusing.

**Comment by SlowMo24** [comment]
> one last question: is the snippet above the full result you get, or is there more which you omitted?

It is the full ohsome-py log output but I have no knowledge if anything gets omitted within ohsome-py

**Comment by SlowMo24** [comment]
The [raw response (see attached file)](https://github.com/GIScience/ohsome-api/files/6575944/outOfMemory.zip) shows that data is transferred for this error just as for a timout that occurs during transaction. Its just not valid json.

**Comment by tyrasd** [comment]
> Its just not valid json.

yes, that's the expected behaviour for extraction endpoints, as [documented](https://github.com/GIScience/ohsome-api/blob/master/docs/http-response-status.rst#2xx-success):

> If an error occurs during a data-extraction request, the result will be a broken GeoJSON containing the error JSON object at the end. In this case, the ohsome API will give back a 200 response status.

**Comment by SlowMo24** [comment]
yes, I just wanted to provide a raw response, as requested, as this was not contained in the errorOnLargeBuffer.zip above (masked by oshome-api).

**Comment by Hagellach37** [comment]
We are moving towards ohsome API v2.0.
Hence, this topic will not be followed up on.

Feel free to raise it again if it is still relevant for the new API version.
